napari/viewer.py

```diff
@@ -61,11 +61,13 @@ def __init__(
         self.window = Window(qt_viewer)
         self.update_console = self.window.qt_viewer.console.push
 
-    def screenshot(self, with_viewer=False):
+    def screenshot(self, path=None, with_viewer=False):
```

Comment:
There's an API change here as a result of adding the `path` kwarg we discussed.

I think `path` should go before the `with_viewer` kwarg because:
1. It keeps this function consistent with all the other screenshot related functions
2. I think use of `path` will be a lot more common than use of `with_viewer`, and I really want people to be able to write: `viewer.screenshot('path/to/screenshot.png')`

Reply:
Are you happy with this, @sofroniewn & @jni?

Reply:
Works for me. Might as well make `with_viewer` keyword-only, which it should have been from the start.

Reply:
Yup, agreed - lets add make it `screenshot(self, path=None, *, with_viewer=False)` so `with_viewer` is keyword-only